File: Practice/11/C++/11. Возведение в степень/11. Возведение в степень/11. Возведение в степень.cpp

#include <iostream>
#include<math.h>

int main()
{
    double a;
    int x;
    int i; i = 1;
    double z;
    setlocale(LC_ALL, "ru");
    std::cout << "Введите число возводимое в степень" << std::endl; std::cin >> a;
    std::cout << "Введите степень" << std::endl; std::cin >> x;
    z = a;
    while (i < abs(x)) {
        a = a * z;
        i++;
    }
    if (x > 0) {
        std::cout << "Результат = " << a << std::endl;
        return 0;
    }
    else if (x < 0) {
        std::cout << "Результат = " << 1/a << std::endl;
        return 0;
    }
    else if (x == 0) {
        std::cout << "Результат = " << 1 << std::endl;
        return 0;
    }
}

File: Practice/11/Python/11. Возведение в степень/11. Возведение в степень/_11._Возведение_в_степень.py
import math
try:
    a=float(input("Введите число возводимое в степень: "))
    x=int(input("Введите степень: "))
    z=1
    z=a
    i=1
    while (i<abs(x)):
        z*=a
        i+=1
    if x>0:
        print("Результат = ",z)
    elif x<0:
        print("Результат = ",1/z)
    elif x==0:
        print("Результат = 1")
except ValueError:
    print("Ошибка ввода.")

